src/main/java/com/example/CleanFolders.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class CleanFolders {
+	private static final List<File> deletableBaks = new ArrayList<>();
+	private static final List<File> folders = new ArrayList<>();
+
+	/**
+	 * Detects directories under the given directory and collects all .bak files without corresponding .doc files next to them.
+	 * The method uses checkFolder() for detection and loneBakDelete() for deletion.
+	 * @param startFolder	File of the starting directory.
+	 */
+	public static void bakDeletion(File startFolder) {
+		checkFolder(startFolder);
+		loneBakDelete();
+	}
+
+	/**
+	 * Returns the collected, deletable .bak files.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of .bak files.
+	 */
+	public static List<File> getDeletableBaks() {
+		return deletableBaks;
+	}
+
+	/**
+	 * Returns the collected directories.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of directories.
+	 */
+	public static List<File> getFolders() {
+		return folders;
+	}
+
+	/**
+	 * Helper method for directory stream filtering and collecting.
+	 * Also, launching the next recursive method call of checkFolder().
+	 * @param file	File of processed folder.
+	 */
+	private static void folderHelper(File file) {
+		folders.add(file);
+		checkFolder(file);
+	}
+
+	/**
+	 * Helper method for file filtering.
+	 * @param file	File of processed file.
+	 * @return		boolean value marks whether the checked file can be deleted.
+	 */
+	private static boolean isDeletableBak(File file) {
+		String path = file.getPath();
+		return file.isFile() &&
+				file.getName().endsWith(".bak") &&
+				!(new File(path.substring(0, path.length() - 4) + ".doc")).exists();
+	}
+
+	/**
+	 * Processes directories and files in a directory.
+	 * Recursive method for directories.
+	 * Puts all directories and only deletable files into correspondent Lists.
+	 * Files with .bak extension and without correspondent .doc file, can be deleted.
+	 * @param folder	File of processed directory.
+	 */
+	private static void checkFolder(File folder) {
+		try {
+			File[] files = folder.listFiles();
+			if ((files != null) && (files.length > 0)) {
+				Arrays.stream(files)
+						.filter(File::isDirectory)
+						.forEach(CleanFolders::folderHelper);
+				Arrays.stream(files)
+						.filter(CleanFolders::isDeletableBak)
+						.forEach(CleanFolders.deletableBaks::add);
+			}
+		}
+		catch (Exception e) {
+			//
+		}
+	}
+
+	/**
+	 * Deletes .bak files without correspondent .doc file next to them, then deletes all empty directories.
+	 */
+	private static void loneBakDelete() {
+		deletableBaks.sort(Comparator.reverseOrder());
+		try {
+			for (File deletableBak : deletableBaks) {
+				if (!deletableBak.delete()) {
+					System.out.println("Unsuccessful deletion: " + deletableBak.getPath());
+				}
+			}
+		}
+		catch (Exception e) {
+			System.out.println("File deletion was terminated due to exception.");
```

Comment:
It worths to include the original exception in the `sout`.

Reply:
I decided to delete the souts from here. The only purpose would be to show if the file system has problems with deletion. The resources are under project root so I think the chance of so problem would be less than 1:million.

Reply:
No, missed java file name. In test, I deleted it, in original class, insert it.
OK